rascaline/src/math/hyp2f1.rs

```diff
@@ -0,0 +1,365 @@
+    use approx::assert_relative_eq;
+
+    #[test]
+    fn test_hyp2f1() {
+        // These  values (a, b, c, x, expected) were computed using
+        // scipy.special.hyp2f1
+        let reference = [
+            (1.0, 2.0, 3.0, 0.0, 1.0),
+            (0.0, 1.0, 0.0, 0.0, 1.0),
+            (0.0, 0.0, 0.0, 0.0, 1.0),
+            (0.0, 0.0, 0.0, -1.0, 1.0),
+            (0.1, 1.0, 2.0, -5.0, 0.8923895138816937),
+            (-2.0, -2.0, -3.0, 0.5, std::f64::INFINITY),
+            (-7.2, -0.5, -4.5, 1e-05, 0.999991999964571),
+            (1.0, 2.0, -3.0, -1.0, std::f64::INFINITY),
+            (1.0, 0.5, -4.5, -20.0, 0.44674682523931875),
+            (-1.0, -2.0, 3.0, 0.0, 1.0),
+            (2.0, -2.0, -3.0, 0.5, std::f64::INFINITY),
+            (2.0, -3.0, -2.0, -1.0, std::f64::INFINITY),
+            (-7.2, 0.5, -4.5, 1e-05, 1.0000080001062877),
+            (-1.0, 2.0, 3.0, 0.0, 1.0),
+            (-1.0, -2.0, 4.0, -1.0, 0.5),
+            (0.0, -6.0, -4.0, -2.0, std::f64::INFINITY),
+            (0.0, -3.0, -4.0, -1.0, std::f64::INFINITY),
+            (1.0, -6.0, -4.0, -1.0, std::f64::INFINITY),
+            (1.0, -0.5, -4.5, -5.0, 0.46975635247240194),
+            (-7.2, -0.5, -4.5, -1.0, 19.915882647295806),
+            (-1.0, -1.0, -2.0, -1.0, std::f64::INFINITY),
+            (-1.0, -1.0, -2.0, 0.0, std::f64::INFINITY),
+            (-1.0, -1.0, -2.0, 0.9, std::f64::INFINITY),
+            (-1.0, -1.0, -1.0, -1.0, std::f64::INFINITY),
+            (-1.0, -1.0, -1.0, 0.0, std::f64::INFINITY),
+            (-1.0, -1.0, -1.0, 0.9, std::f64::INFINITY),
+            (-1.0, -1.0, 0.0, 0.0, 1.0),
+            (-1.0, 0.0, -2.0, -1.0, std::f64::INFINITY),
+            (-1.0, 0.0, -2.0, 0.0, std::f64::INFINITY),
+            (-1.0, 0.0, -2.0, 0.9, std::f64::INFINITY),
+            (-1.0, 0.0, -1.0, -1.0, std::f64::INFINITY),
+            (-1.0, 0.0, -1.0, 0.0, std::f64::INFINITY),
+            (-1.0, 0.0, -1.0, 0.9, std::f64::INFINITY),
+            (-1.0, 0.0, 0.0, -1.0, 1.0),
+            (-1.0, 0.0, 0.0, 0.0, 1.0),
+            (-1.0, 0.0, 0.0, 0.9, 1.0),
+            (-1.0, 1.0, -2.0, -1.0, std::f64::INFINITY),
+            (-1.0, 1.0, -2.0, 0.0, std::f64::INFINITY),
+            (-1.0, 1.0, -2.0, 0.9, std::f64::INFINITY),
+            (-1.0, 1.0, -1.0, -1.0, std::f64::INFINITY),
+            (-1.0, 1.0, -1.0, 0.0, std::f64::INFINITY),
+            (-1.0, 1.0, -1.0, 0.9, std::f64::INFINITY),
+            (-1.0, 1.0, 0.0, 0.0, 1.0),
+            (0.0, -1.0, -2.0, -1.0, std::f64::INFINITY),
+            (0.0, -1.0, -2.0, 0.0, std::f64::INFINITY),
+            (0.0, -1.0, -2.0, 0.9, std::f64::INFINITY),
+            (0.0, -1.0, -1.0, -1.0, std::f64::INFINITY),
+            (0.0, -1.0, -1.0, 0.0, std::f64::INFINITY),
+            (0.0, -1.0, -1.0, 0.9, std::f64::INFINITY),
+            (0.0, -1.0, 0.0, -1.0, 2.0 / 2.0),
+            (0.0, -1.0, 0.0, 0.0, 1.0),
+            (0.0, -1.0, 0.0, 0.9, 0.09999999999999998 * 10.),
```

Comment:
There is still some work todo. All values where I multiplied something do not agree with the scipy version. We have to take a look at these.

Reply:
oh, I did not get that part

Reply:
I also did not mention it.

Reply:
Scipy might also be wrong here. 

```
import mpmath
import scipy.special


a = 0.0
b = 1.0
c = 0.0
x = 0.9

print("mpmath:", mpmath.hyp2f1(a, b, c, x))
print("scipy:", scipy.special.hyp2f1(a, b, c, x))
```

```
mpmath: 1.0
scipy: 10.000000000000002
```

Reply:
Thanks for double checking!